src/openrct2/actions/CheatSetAction.cpp

```diff
@@ -512,13 +512,23 @@ void CheatSetAction::FixBrokenRides() const
 {
     for (auto& ride : GetRideManager())
     {
-        if ((ride.mechanic_status != RIDE_MECHANIC_STATUS_FIXING)
-            && (ride.lifecycle_flags & (RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN)))
+        if (ride.lifecycle_flags & (RIDE_LIFECYCLE_BREAKDOWN_PENDING | RIDE_LIFECYCLE_BROKEN_DOWN))
         {
             auto mechanic = RideGetAssignedMechanic(ride);
-            if (mechanic != nullptr)
+
+            if (ride.mechanic_status == RIDE_MECHANIC_STATUS_FIXING)
             {
-                mechanic->RemoveFromRide();
+                if (mechanic != nullptr)
+                {
+                    mechanic->RideSubState = PeepRideSubState::ApproachExitWaypoints;
+                }
+            }
+            else
```

Comment:
We’ll probably want to limit this `else` to "responding to breakdown call" and "answering breakdown call". It prevents mechanics heading for an inspection from being unceremoniously kicked from the job. (I know that that’s what it’s already doing on develop, but this seemed like a good opportunity to change it.)

Reply:
> We’ll probably want to limit this `else` to "responding to breakdown call" and "answering breakdown call". It prevents mechanics heading for an inspection from being unceremoniously kicked from the job. (I know that that’s what it’s already doing on develop, but this seemed like a good opportunity to change it.)

Yeah that is a good point, we don't want mechanics inspecting or on the way to inspect to be removed from the ride. Thank you.